docs/contributing/Regular-Expression-Regex.html

```diff
@@ -0,0 +1,1140 @@
+  </tr>
+  <tr>
+    <td>havok_fix.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/91598/">(SKSE) Havok Fix</a></td>
+  </tr>
+  <tr>
+    <td>LipSyncFix.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/75951/">Fix Lip Sync</a></td>
+  </tr>
+  <tr>
+    <td>LocationalDamage.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/91505/">Locational Damage</a></td>
+  </tr>
+  <tr>
+    <td>nioverride.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/29624/">RaceMenu</a></td>
+  </tr>
+  <tr>
+    <td>PlayerRotation.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/102937/">Player Rotation in ShowRaceMenu</a></td>
+  </tr>
+  <tr>
+    <td>po3_namedashpiles.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/97046/">Name Those Ash Piles</a></td>
+  </tr>
+  <tr>
+    <td>po3_papyrusextender.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/95017/">powerofthree's Papyrus Extender</a></td>
+  </tr>
+  <tr>
+    <td>po3_SpellPerkItemDistributor.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/105121/">Spell Perk Item Distributor</a></td>
+  </tr>
+  <tr>
+    <td>po3_WashThatBloodOff.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/98546/">Wash That Blood Off</a></td>
+  </tr>
+  <tr>
+    <td>QuickLoot.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/73096/">Quick Loot</a></td>
+  </tr>
+  <tr>
+    <td>SkyrimSouls.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/83379/">SkyrimSouls - Unpaused Game Menus</a></td>
+  </tr>
+  <tr>
+    <td>StorageUtil.dll</td>
+    <td><a href="https://www.nexusmods.com/skyrim/mods/64798/">Campfire - Complete Camping System</a></td>
+  </tr>
+</table> 
+
+<p>Summary:</p>
+
+<ul>
+  <li>The <code>condition</code> in this example includes the regular expression: <code>file("SKSE/Plugins/([^\.]+\.dll)")</code></li>
+  <li>The objective is to find <code>.dll</code> files within the folder structure <code></code>SKSE/Plugins/</li>
+  <li>As a first step the period <code>.</code> in front of <code>dll</code> has been escaped through writing a backslash <code>\</code> in front of it.</li>
+  <li>The remaining part of the regex <code>[^\.]+</code> consists out of three parts.</li>
+  <li><code>[^x]</code> will find a character that is <i>not x</i>, which is achieved by placing the circumflex <code>^</code> within square brackets <code>[ ]</code>. In this case <code>x</code> is an escaped period <code>\.</code> so <i>not a period</i> is what we want. After the square brackets comes the quantifier plus <code>+</code> which means <i>one or more times</i>.</li>
+  <li>Therefore the regular expression <code>[^\.]+\.dll</code> finds all SKSE plugins, which filesnames include one or more characters that are not a period, and end with <code>.dll</code></li>
+</ul>
+
+
+<h2>Lookarounds</h2>
+
+<h3>Negative lookahead: <code>(?! ...)</code></h3>
```

Comment:
Like the Perl character classes at the start, support for lookaround varies by implementation and I think in LOOT negative lookahead is supported everywhere apart from inside conditions.

Reply:
I've added a notice in https://github.com/loot/loot.github.io/pull/109/commits/0e54156b2c3af237cf3db7c585e1bdef3d531e19.